test/common/access_log/access_log_impl_test.cc

```diff
@@ -918,6 +921,111 @@ name: envoy.file_access_log
+      "ok",
+      "canceled",
+      "unknown",
+      "invalid_argument",
+      "deadline_exceeded",
+      "not_found",
+      "already_exists",
+      "permission_denied",
+      "resource_exhausted",
+      "failed_precondition",
+      "aborted",
+      "out_of_range",
+      "unimplemented",
+      "internal",
+      "unavailable",
+      "data_loss",
+      "unauthenticated",
+  });
+
+  std::ostringstream out;
+  out << R"EOF(
+name: envoy.file_access_log
+filter:
+  grpc_status_filter:
+    statuses:
+)EOF";
+
+  for (const auto& valid_status : valid_statuses) {
+    out << "      - " << valid_status << std::endl;
+  }
+
+  out << R"EOF(config:
+  path: /dev/null
+  )EOF";
+
+  // We expect that if any of the statuses in valid_statuses are invalid, that this line will throw
+  // an error.
+  AccessLogFactory::fromProto(parseAccessLogFromV2Yaml(out.str()), context_);
+}
+
+TEST_F(AccessLogImplTest, GrpcStatusFilterUnsupportedValue) {
+  const std::string yaml = R"EOF(
+name: envoy.file_access_log
+filter:
+  grpc_status_filter:
+    statuses:
+      - not_a_valid_code
+config:
+  path: /dev/null
+  )EOF";
+
+  EXPECT_THROW_WITH_MESSAGE(
+      AccessLogFactory::fromProto(parseAccessLogFromV2Yaml(yaml), context_),
+      ProtoValidationException,
+      "Proto constraint validation failed (AccessLogFilterValidationError.GrpcStatusFilter: "
+      "[\"embedded message failed validation\"] | caused by "
+      "GrpcStatusFilterValidationError.Statuses[i]: [\"value must be in list \" [\"ok\" "
+      "\"canceled\" \"unknown\" \"invalid_argument\" \"deadline_exceeded\" \"not_found\" "
+      "\"already_exists\" \"permission_denied\" \"resource_exhausted\" \"failed_precondition\" "
+      "\"aborted\" \"out_of_range\" \"unimplemented\" \"internal\" \"unavailable\" \"data_loss\" "
+      "\"unauthenticated\"]]): grpc_status_filter {\n  statuses: \"not_a_valid_code\"\n}\n");
+}
+
+TEST_F(AccessLogImplTest, GrpcStatusFilterPass) {
+  const std::string yaml = R"EOF(
+name: envoy.file_access_log
+filter:
+  grpc_status_filter:
+    statuses:
+      - ok
+config:
+  path: /dev/null
+  )EOF";
+
+  const InstanceSharedPtr log =
+      AccessLogFactory::fromProto(parseAccessLogFromV2Yaml(yaml), context_);
+
+  response_trailers_.addCopy(Http::Headers::get().GrpcStatus, "0");
+
+  EXPECT_CALL(*file_, write(_));
+  log->log(&request_headers_, &response_headers_, &response_trailers_, stream_info_);
+}
+
+TEST_F(AccessLogImplTest, GrpcStatusFilterBlock) {
+  const std::string yaml = R"EOF(
+name: envoy.file_access_log
+filter:
+  grpc_status_filter:
+    statuses:
+      - ok
+config:
+  path: /dev/null
+  )EOF";
+
+  const InstanceSharedPtr log =
+      AccessLogFactory::fromProto(parseAccessLogFromV2Yaml(yaml), context_);
+
+  response_trailers_.addCopy(Http::Headers::get().GrpcStatus, "1");
+
+  EXPECT_CALL(*file_, write(_)).Times(0);
+  log->log(&request_headers_, &response_headers_, &response_trailers_, stream_info_);
+}
+
```

Comment:
Can you add tests to show the defaulting to HTTP response codes?

Reply:
Will write and submit another commit tomorrow.